### PR TITLE
changed: use compressed field properties setting up numbers

### PR DESCRIPTION
**Pull request by akva2**
Let's go slow here. Split out from https://github.com/OPM/opm-simulators/pull/2374

### CODE REVIEW

**Comment by akva2** [comment]
jenkins run norne_parallel please

**Comment by akva2** [comment]
i hope to use your fever to my advantage ;P

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -2850,32 +2850,22 @@ private:
...
-        const auto& numData = get_int_prop(eclState, name);
+        const auto& numData = eclState.fieldProps().get_int(name);
         const auto& vanguard = simulator.vanguard();
 
         unsigned numElems = vanguard.gridView().size(/*codim=*/0);
         numbers.resize(numElems);
         for (unsigned elemIdx = 0; elemIdx < numElems; ++elemIdx) {
-            unsigned cartElemIdx = vanguard.cartesianIndex(elemIdx);
-            numbers[elemIdx] = static_cast<T>(std::max(numData[cartElemIdx], 1) - 1);
+            numbers[elemIdx] = numData[elemIdx] -1;
```
Comment (blattms) [inline ebos/eclproblem.hh]: That change I do not get.
If taking `std::max` was necessary before, why isn't now, too. The mapping should element to numData should be the same, no?

Or was it superfluous before, too?

Comment (akva2) [reply ebos/eclproblem.hh]: it was always superfluous. bska was overly concerned with non-active cells, i did not fight it back when i had to introduce it. i can keep it but it serves no role (for valid data).

Comment (akva2) [reply ebos/eclproblem.hh]: that being said i should keep the cast.

**Pull request merged**